### PR TITLE
Wrap grdlandmask

### PR DESCRIPTION
**Pull request by willschlitzer**
This pull request wraps the [`grdlandmask`](https://docs.generic-mapping-tools.org/6.1/grdlandmask.html) module.

Fixes #

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by willschlitzer** [comment]
I'm running into a newfound problem with this module. When I run:
```
import pygmt
new_grid = pygmt.grdlandmask(increment=1, region = [-5, 5, -5, 5])
```

I receive: `TypeError: 'module' object is not callable`

I've done the typical trouble shooting (confirmed I'm in the correct conda environment, have the appropriate git branch activated, checked the import statements and function to make sure it's in line with other modules, and, of course, shut it down and started it up again) so I'm not sure why I'm getting this issue. @weiji14 @seisman any ideas?

File: pygmt/src/__init__.py
Comment (weiji14) [inline pygmt/src/__init__.py:18]: Typo here.
```suggestion
from pygmt.src.grdlandmask import grdlandmask
```

File: pygmt/src/grdlandmask.py
Comment (weiji14) [inline pygmt/src/grdlandmask.py:23]:
```suggestion
def grdlandmask(**kwargs):
```

File: pygmt/src/__init__.py
Comment (willschlitzer) [reply pygmt/src/__init__.py:18]: :facepalm:

File: pygmt/src/grdlandmask.py
Comment (seisman) [inline pygmt/src/grdlandmask.py:77]:
```suggestion
            raise GMTInvalidInput("Region and increment must be specified.")
```

Comment (seisman) [inline pygmt/src/grdlandmask.py:77]: It's unnecessary to create a temporary file when the two parameters are not given. So, moving the checking outside of the `with` statement.
```suggestion
    """
    if "I" not in kwargs.keys() and "R" not in kwargs.keys():
        raise GMTInvalidInput("Region and increment must be specified.")
        
    with GMTTempFile(suffix=".nc") as tmpfile:
```

File: pygmt/src/grdlandmask.py
Comment (seisman) [inline pygmt/src/grdlandmask.py:21]:
```suggestion
    I="increment",
    r="registration",
```

File: pygmt/src/grdlandmask.py
Comment (seisman) [inline pygmt/src/grdlandmask.py:76]: I don't test it, but it seems both **-R** and **-I** are required for this module. If yes, we should change it to 
```
if "I" not in kwargs.keys() or "R" not in kwargs.keys():
```

Comment (willschlitzer) [reply pygmt/src/grdlandmask.py:76]: Yep; I screwed up the logic on that one!

Comment (willschlitzer) [reply pygmt/src/grdlandmask.py]: Fixed in aaff96a

File: pygmt/src/grdlandmask.py
Comment (seisman) [inline pygmt/src/grdlandmask.py:65]:
```suggestion
    {R}
    {r}
```

**Review by seisman — APPROVED** [review]
Looks good to me.

**Review by michaelgrund — APPROVED** [review]

File: pygmt/tests/test_grdlandmask.py
Comment (core-man) [inline pygmt/tests/test_grdlandmask.py:14]:
```suggestion
    Creates a grid land mask with an outgrid argument.
```

**Review by core-man — COMMENTED** [review]
Great job. Just one minor comment.

**Review by core-man — APPROVED** [review]

File: pygmt/src/grdlandmask.py
Comment (meghanrjones) [inline pygmt/src/grdlandmask.py:21]: `blockmean`, `blockmedian`, `grdfilter`, `surface` and `x2sys_init` all use `spacing` for the same usage, so I recommend using `spacing` rather than `increment` here.
```suggestion
    I="spacing",
```

File: pygmt/src/grdlandmask.py
Comment (meghanrjones) [inline pygmt/src/grdlandmask.py:64]: This would be a good option to include as standardized text, similar to https://github.com/GenericMappingTools/pygmt/pull/1186. Here is a list of all the GMT modules that use identical text to describe this parameter, so PyGMT could similarly use standardized text in its docs.
```
blockmean
blockmedian
blockmode
gmtbinstats
grdblend
grdfilter
grdlandmask
grdmask
grdmath
grdsample
mask_common_
nearneighbor
sph2grd
sphdistance
sphinterpolate
supplements/geodesy/earthtide
supplements/geodesy/gpsgridder
supplements/potential/grdgravmag3d
supplements/potential/grdseamount
supplements/potential/talwani3d
supplements/spotter/grdspotter
supplements/spotter/hotspotter
supplements/spotter/polespotter
surface
triangulate
xyz2grd
```

File: pygmt/src/grdlandmask.py
Comment (meghanrjones) [inline pygmt/src/grdlandmask.py:45]:
```suggestion
    spacing : str
```

File: pygmt/tests/test_grdlandmask.py
Comment (meghanrjones) [inline pygmt/tests/test_grdlandmask.py:17]:
```suggestion
        result = grdlandmask(outgrid=tmpfile.name, spacing=1, region=[-5, 5, -5, 5])
```

File: pygmt/tests/test_grdlandmask.py
Comment (meghanrjones) [inline pygmt/tests/test_grdlandmask.py:30]:
```suggestion
    temp_grid = grdlandmask(spacing=1, region=[-5, 5, -5, 5])
```

File: pygmt/src/grdlandmask.py
Comment (meghanrjones) [inline pygmt/src/grdlandmask.py:78]:
```suggestion
        raise GMTInvalidInput("Region and spacing must be specified.")
```

File: pygmt/src/grdlandmask.py
Comment (weiji14) [reply pygmt/src/grdlandmask.py:64]: I wouldn't put this huge block of text for spacing (-I) in `fmt_docstring` like #1186 to be honest, but maybe we could add it to the COMMON_OPTIONS dictionary in [decorators.py](https://github.com/GenericMappingTools/pygmt/blob/master/pygmt/helpers/decorators.py) (even if it isn't exactly a GMT common option)?

Comment (meghanrjones) [reply pygmt/src/grdlandmask.py:64]: These seems like a good option to me. There are already multiple non-common options in COMMON_OPTIONS anyways.

Comment (weiji14) [reply pygmt/src/grdlandmask.py:64]: Ok, I'll work on a PR for this after lunch.

Comment (meghanrjones) [reply pygmt/src/grdlandmask.py:64]: Sorry for slowing you down. If only we had more online hours in common.

**Review by meghanrjones — APPROVED** [review]

**Comment by seisman** [comment]
@willschlitzer Feel free to merge this PR after all tests pass.

File: pygmt/src/grdlandmask.py
Comment (seisman) [inline pygmt/src/grdlandmask.py:64]: Perhaps we should wait for #1288 before merging this PR.

Comment (weiji14) [reply pygmt/src/grdlandmask.py:64]: Agree, it shouldn't take too long. And once that's done, we can replace the block of text with this:

```suggestion
    {I}
```

**Review by weiji14 — CHANGES_REQUESTED** [review]

File: pygmt/src/grdlandmask.py
Comment (weiji14) [inline pygmt/src/grdlandmask.py:21]:
```suggestion
    I="spacing",
    R="region",
```

File: pygmt/tests/test_grdlandmask.py
Comment (weiji14) [inline pygmt/tests/test_grdlandmask.py:33]:
```suggestion
    assert temp_grid.gmt.registration == 0  # Pixel registration
```

File: pygmt/tests/test_grdlandmask.py
Comment (weiji14) [inline pygmt/tests/test_grdlandmask.py:40]:
```suggestion
    Check that grdlandmask fails correctly when region and spacing are not
    given.
```

File: pygmt/src/grdlandmask.py
Comment (weiji14) [inline pygmt/src/grdlandmask.py:59]:
```suggestion
        raise GMTInvalidInput("Both 'region' and 'spacing' must be specified.")
```

**Review by weiji14 — APPROVED** [review]
Just some final suggestions. Feel free to merge after the tests pass.

**Pull request merged**
